### PR TITLE
[IT-607] Setup Security Hub

### PR DESCRIPTION
**Pull request by zaro0508**
Setup security hub with notifications sent to endpoints.
Multiple protocols and endpoints can be configured.

This template is based on templates provided by
https://asecure.cloud/p/config_securityhub

### CODE REVIEW

**Comment by zaro0508** [comment]
this PR generates the following templates:

<detail>
<summary>us-east-1-security-central-account.yaml</summary>
    AWSTemplateFormatVersion: '2010-09-09'
    Description: SecurityHub notifications
    Parameters: {}
    Resources:
      SecurityHub:
        Type: AWS::SecurityHub::Hub
    Outputs:
      SecurityHubArn:
        Description: The security hub ARN
        Value:
          Ref: SecurityHub
        Export:
          Name:
            Fn::Sub: ${AWS::Region}-${AWS::StackName}-SecurityHubArn
</detail>

<detail>
<summary>us-east-1-security-central-account.yaml</summary>
    AWSTemplateFormatVersion: '2010-09-09'
    Description: SecurityHub notifications
    Parameters: {}
    Resources:
      SecurityHubTopic:
        Type: AWS::SNS::Topic
        Properties:
          Subscription:
            - Protocol: email
              Endpoint: foo@sagebase.org
            - Protocol: email
              Endpoint: bar@sagebase.org
          TopicName: security-hub-topic
      CloudwatchEvent:
        Type: AWS::Events::Rule
        Properties:
          Name: detect-securityhub-finding
          Description: >-
            A CloudWatch Event Rule that triggers on AWS Security Hub findings. The
            Event Rule can be used to trigger notifications or remediative actions
            using AWS Lambda."
          State: ENABLED
          Targets:
            - Arn:
                Ref: SecurityHubTopic
              Id: target-id1
          EventPattern:
            detail-type:
              - Security Hub Findings - Imported
            source:
              - aws.securityhub
      SnsTopicPolicy:
        Type: AWS::SNS::TopicPolicy
        Properties:
          PolicyDocument:
            Statement:
              - Sid: __default_statement_ID
                Effect: Allow
                Principal:
                  AWS: '*'
                Action:
                  - SNS:GetTopicAttributes
                  - SNS:SetTopicAttributes
                  - SNS:AddPermission
                  - SNS:RemovePermission
                  - SNS:DeleteTopic
                  - SNS:Subscribe
                  - SNS:ListSubscriptionsByTopic
                  - SNS:Publish
                  - SNS:Receive
                Resource:
                  Ref: SecurityHubTopic
                Condition:
                  StringEquals:
                    AWS:SourceOwner:
                      Ref: AWS::AccountId
              - Sid: TrustCWEToPublishEventsToMyTopic
                Effect: Allow
                Principal:
                  Service: events.amazonaws.com
                Action: sns:Publish
                Resource:
                  Ref: SecurityHubTopic
          Topics:
            - Ref: SecurityHubTopic
    Outputs:
      SecurityHubTopicArn:
        Description: The security hub topic ARN
        Value:
          Ref: SecurityHubTopic
        Export:
          Name:
            Fn::Sub: ${AWS::Region}-${AWS::StackName}-SecurityHubTopicArn
</detail>

File: org-formation/075-security-hub/_tasks.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  StackName: !Sub '${resourcePrefix}-${appName}-notifications'
+  StackDescription: SecurityHub notifications
+  DefaultOrganizationBindingRegion: !Ref primaryRegion
+  DefaultOrganizationBinding:
+    Account: !Ref accountId
+  TemplatingContext:
+    Subscriptions:
+      # refer to https://docs.aws.amazon.com/sns/latest/api/API_Subscribe.html for valid list of protocol and endpoints
+      - Protocol: "email"
+        Endpoints: ["foo@sagebase.org","bar@sagebase.org"]
```
Comment (ahayden) [inline org-formation/075-security-hub/_tasks.yaml:34]: Can you parameterize this and input as a var from the workflow?

**Review by ahayden — APPROVED** [review]

File: org-formation/075-security-hub/_tasks.yaml (hunk 1 above)
Comment (zaro0508) [reply org-formation/075-security-hub/_tasks.yaml:34]: `Protocol` and `Endpoints` are vars.  These vars get passed into `notifications.njk` template,  you just need to decide what emails you want here.

Comment (ahayden) [reply org-formation/075-security-hub/_tasks.yaml:34]: probably `it-alert-inbox@`

**Review by ahayden — APPROVED** [review]

**Pull request merged**